src/cache/cache_las.c

```diff
@@ -611,8 +611,17 @@ __wt_las_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_PAGE *page, WT_MU
     /* Wrap all the updates in a transaction. */
     WT_ERR(__wt_txn_begin(session, NULL));
     __las_set_isolation(session, &saved_isolation);
+
     local_txn = true;
 
+    /*
+     * Set the oldest timestamp as the first commit timestamp for the LAS transaction. None of the
+     * transactions with commit timestamp less than oldest timestamp will go into history store. AS
+     * they will be cleaned as part of the obsolete check.
```

Comment:
The assumption "None of the transactions with commit timestamp less than oldest timestamp will go into history store" is wrong. The version visible at oldest timestamp will go to history store, the version before that one can be discarded.

Reply:
The code was not removing all the versions older than the oldest and leaving one visible record. The comment was wrong. Removed it.